rewrite-json/src/main/java/org/openrewrite/json/CopyValue.java

```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright 2020 the original author or authors.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.json;
+
+import lombok.Data;
+import lombok.EqualsAndHashCode;
+import lombok.Value;
+import org.jspecify.annotations.Nullable;
+import org.openrewrite.ExecutionContext;
+import org.openrewrite.FindSourceFiles;
+import org.openrewrite.Option;
+import org.openrewrite.Preconditions;
+import org.openrewrite.ScanningRecipe;
+import org.openrewrite.SourceFile;
+import org.openrewrite.TreeVisitor;
+import org.openrewrite.json.tree.Json;
+
+import java.nio.file.Path;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+@SuppressWarnings("LanguageMismatch")
+@Value
+@EqualsAndHashCode(callSuper = false)
+public class CopyValue extends ScanningRecipe<CopyValue.Accumulator> {
+    @Option(displayName = "Old key path",
+            description = "A [JsonPath](https://docs.openrewrite.org/reference/jsonpath-and-jsonpathmatcher-reference) expression to locate a JSON key/value pair to copy.",
+            example = "$.source.kind")
+    String oldKeyPath;
+
+    @Option(displayName = "Old file path",
+            description = "The file path to the JSON file to copy the value from. " +
+                          "If `null` then the value will be copied from any JSON file it appears within.",
+            example = "src/main/resources/application.json",
+            required = false)
+    @Nullable
+    String oldFilePath;
+
+    @Option(displayName = "New key path",
+            description = "A [JsonPath](https://docs.openrewrite.org/reference/jsonpath-and-jsonpathmatcher-reference) expression defining where the value should be written.",
+            example = "$.dest.kind")
+    String newKey;
```

Comment:
According to `AddKeyValue` this parameter is just a key name, not a path as the description of this parameter suggests.